Fix adding network IP other than WireGuard IP

Mahmoud-Emad opened the pull request:
### Description

As @AhmedHanafy725  reported, there was an issue when selecting another network other than the WireGuard IP

```log
WorkloadDeployError: Failed to deploy gateway-name-proxy with name vm26vmf1d6qf on node 11 due to: failed to setup name proxy: failed to get user network: couldn't load network with id (pL1JFSUwD5VFC): open /var/run/cache/networkd/networks/pL1JFSUwD5VFC: no such file or directory.
```
So, to avoid this error, we must send the network option field only if the WireGuard IP selected


### Changes

- Set the network field as an optional field
- Re-order the list of the networks
- Set the 'WireGuard IP' as the default IP
- Remove the subnet from the IP on change
- Lowercase the subdomain to avoid the validation issues

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3148
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3232


### Documentation PR

For UI changes, Please provide the Documetation PR on [info_grid](https://github.com/threefoldtech/info_grid)

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

Mahmoud-Emad committed:
fix: Fix adding domain to deployed machine:

- Set the network field as an optional field
- Re-order the list of the networks
- Set the 'WireGuard IP' as the defualt IP
- Remove the subnet from the IP on change

Mahmoud-Emad committed:
fix: Fix the SubDomian input validation

- Lowercase the subdomain to avoid the validation issues

Mahmoud-Emad committed:
fix: Fix Playground build by importing 'DeployGatewayConfig' as type.

AhmedHanafy725 commented on packages/playground/src/components/manage_gateway_dialog.vue:
```diff
@@ -240,7 +254,7 @@ export default {
         grid.config.twinId;
       prefix.value = oldPrefix.value + props.vm.name;
-      subdomain.value = generateName({ prefix: prefix.value }, 4);
+      subdomain.value = generateName({ prefix: prefix.value }, 4).toLowerCase();
```
the vm's name might have `_`, so will it be valid after the lowercase?

Mahmoud-Emad replied:
The issue arises because the VM name can contain underscores, but domain names cannot. Previously, we didn’t check for underscores when generating a domain name based on the VM name.

I added a function that suggests a subdomain name by checking for underscores in the VM name and replacing them with an empty string.

Good catch!

Mahmoud-Emad committed:
fix: ensure valid subdomain names by removing underscores from VM names

- Added a function to check for underscores in VM names when generating domain names.
- Replaced underscores with empty strings to ensure that the generated subdomain names are valid.

0oM4R commented on packages/playground/src/components/manage_gateway_dialog.vue:
```diff
@@ -235,12 +249,7 @@ export default {
-        grid.config.twinId;
-      prefix.value = oldPrefix.value + props.vm.name;
-      subdomain.value = generateName({ prefix: prefix.value }, 4);
+      suggestName();
```
I'm wondering if we can avoid this warning 

[Screencast from 2024-08-12 18-33-45.webm](https://github.com/user-attachments/assets/71a75c00-8a7c-4102-95c4-2b54cb38ef57)


after deploying the domain, and i want to deploy a new domain, the suggested name is still as the previous one, maybe we can use computed ref to suggest new name on opening tab1

Mahmoud-Emad committed:
fix: Suggest another domain name after successful deployment

0oM4R reviewed (approved):
![Screenshot from 2024-08-13 12-16-36](https://github.com/user-attachments/assets/7cdcd2ec-dcae-4831-ad57-1206b5fa5708)
tested, 
also test each ip separately

AhmedHanafy725 reviewed (commented):
can we change the `supported ips` to `supported interfaces`? and it should be before the network name (we may hide the network name entirely)

Mahmoud-Emad committed:
style: Move the Network name input into the end:

- Moved the Network name input into the end of the form.
- Renamed the 'Supported Ips' to 'Supported interfaces'.
- Rename the 'NetworkIP' enum to 'NetworkInterfaces'.
